pyccel/parser/semantic.py

```diff
@@ -2682,6 +2688,37 @@ def _visit_DottedName(self, expr):
             bounding_box=(self.current_ast_node.lineno, self.current_ast_node.col_offset),
             severity='fatal')
 
+    def _visit_ListExtend(self, list_variable, iterable):
+        pyccel_stage.set_stage('syntactic')
+
+        if isinstance(iterable, PythonList):
+            store = []
+            for i in iterable.args:
+                arg = FunctionCallArgument(i)
+                func_call = FunctionCall('append', [arg])
+                dotted = DottedName(list_variable, func_call)
+                lhs = PyccelSymbol('_', is_temp=True)
+                assign = Assign(lhs, dotted)
+                assign.set_current_ast(iterable.python_ast)
+                store.append(assign)
+            result = CodeBlock(store)
+            pyccel_stage.set_stage('semantic')
+            return self._visit(result)
```

Comment:
It is useful to create syntactic nodes when the semantic handling is complicated and would result in a lot of code duplication. I am not convinced that this is the case here. Can you not just do:
```suggestion
        if isinstance(iterable, PythonList):
            var = self._visit(list_variable)
            added_list = self._visit(iterable)
            store = [ListAppend(var, a) for a in added_list]
            return CodeBlock(store)
```
?